ExtraDry/ExtraDry.Server/Internal/StaticServiceProvider.cs

```diff
@@ -0,0 +1,18 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+
+namespace ExtraDry.Server.Internal
+{
+    /// <summary>
+    /// Provides access to the service provider for access from static extension methods. 
+    /// Prefer use of established DI principles over this. 
+    /// </summary>
+    internal static class StaticServiceProvider
+    {
+        public static void Initialize(IServiceCollection serviceCollection)
+        {
+            Provider ??= serviceCollection.BuildServiceProvider();
```

Comment:
I don't think this will work here, you're not getting the global service provider, you're building an intermediate service provider that only has the benefit of the services that have been registered up to the point of AddExtraDry.

Reply:
I'll delete this and use the other approach discussed